dcore/advection.py

```diff
@@ -185,7 +190,7 @@ def apply(self, x_in, x_out):
 
 class EmbeddedDGAdvection(Advection):
 
-    def __init__(self, state, V, Vdg, continuity):
+    def __init__(self, state, Vdg, continuity):
 
         super(EmbeddedDGAdvection, self).__init__(state)
         self.dgadvection = DGAdvection(state, Vdg, continuity)
```

Comment:
While you're here, I think you should inherit from DGAdvection and then you can simplify things:

```python
class EmbeddedDGAdvection(DGAdvection):
    def __init__(self, state, Vdg, continuity):
        super(EmbeddedDGAdvection, self).__init__(state, Vdg, continuity)
        self.xdg_in = Function(Vdg)
        self.xdg_out = Function(Vdg)

    def apply(self, x_in, x_out):
        self.xdg_in.interpolate(x_in)
        # Call normal DG advection on DG fields via superclass
        super(EmbeddedDGAdvection, self).apply(self.xdg_in, self.xdg_out)
        x_out.project(self.xdg_out)
```

That way you're not carrying around an extra ubar that you need to move from this advection object to the dgadvection object inside it.